backend/Shared/HitMeApp.Shared.Infrastructure/Integration/IntegrationEvent.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System;
+
+namespace HitMeApp.Shared.Infrastructure.Integration
+{
+    public abstract class IntegrationEvent
+    {
+        public Guid Id { get; }
+        public DateTime IssuedAt { get; }
+
+        protected IntegrationEvent()
+        {
+            Id = Guid.NewGuid();
+            IssuedAt = DateTime.UtcNow;
+        }
```

Comment:
Could this constructor call second one to initialize these properties?

Reply:
Oh yeah, it sure can.